### PR TITLE
Add sanity check to disk space check

### PR DESCRIPTION
**Pull request by F1ko**
Fixes https://github.com/kubernetes/minikube/issues/12761

### CODE REVIEW

**Comment by F1ko** [comment]
This PR is identical to PR https://github.com/kubernetes/minikube/pull/12814 which acted out for some reason.

@spowelljr labeld the original PR with `/ok-to-test`.

Would be nice to pick up where we left off.

**Comment by F1ko** [comment]
Looking through the git history I observed that workflow failures seem to be the case quite often, even for already merged commits. Does anyone mind explaining to me how to interpret that correctly? Are these certain "know-issues" and expected to fail? In that case, how would I know whether this is due to my changes or because it indeed is a known issue?

**Comment by sharifelgamal** [comment]
We do unfortunately have some known test failures, and other tests that are simply flaky, that fail for reasons unrelated to your PR. Once all the tests have run, we have a bot that will comment the flake rate of all the failed tests. If there are tons of 0% tests, then it's a safe bet that your PR broke something. Otherwise, maintainers just use their best judgement when merging PRs.

**Comment by F1ko** [comment]
hi @spowelljr 

are the flake rates high enough now or should I look into other issues as well?
any advice on debugging hyper-v windows without a native windows machine?

File: pkg/minikube/machine/start_test.go [hunk 1]
```diff
@@ -96,3 +98,102 @@ func writeContentToTempFile(content string) (string, error) {
...
+		},
+	}
+	for _, tt := range tests {
+		t.Logf("starting %v", tt.name)
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := DiskUsed(tt.args.cr, tt.args.dir)
+			t.Logf("err: %v\n", err)
+			t.Logf("got: %v\n", got)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("DiskUsed() error = %v, wantErr %v", err, tt.wantErr)
```
Comment (spowelljr) [inline pkg/minikube/machine/start_test.go:142]: If you don't want execution to continue you can do `t.Fatalf` instead and then you don't have to add the return yourself.

File: pkg/minikube/machine/start_test.go [hunk 2]
```diff
@@ -96,3 +98,102 @@ func writeContentToTempFile(content string) (string, error) {
...
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Logf("starting %v", tt.name)
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := DiskAvailable(tt.args.cr, tt.args.dir)
+			t.Logf("err: %v\n", err)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("DiskAvailable() error = %v, wantErr %v", err, tt.wantErr)
```
Comment (spowelljr) [inline pkg/minikube/machine/start_test.go:191]: Same as above

**Review by spowelljr — CHANGES_REQUESTED** [review]
Just a small thing and then it's good to merge

**Comment by F1ko** [comment]
@spowelljr thanks, fixed

**Review by spowelljr — APPROVED** [review]

**Pull request merged**